### PR TITLE
fix: Revert to get_checkpoints.sql call to enable NaN & Infinity values in searcher metric

### PR DESCRIPTION
**Pull request by AmanuelAaron**
## Ticket
ET-156

## Description

Bun unmarshalling of JSON does not support `NaN`, `Infinity` & `-Infinity` values

## Test Plan

Unit tests should pass.
Create checkpoints with a `NaN`, `Infinity` & `-Infinity` values, similar to the ones in the added test and call `GetCheckpoint`. It should return with the correct values in both `searcher_metric` & `validation_metrics`.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: master/internal/api_checkpoint_intg_test.go [hunk 1]
```diff
@@ -328,6 +329,104 @@ func TestCheckpointReturned(t *testing.T) {
...
+					TrialId:        int32TrialID,
+					TrialRunId:     0,
+					StepsCompleted: ptrs.Ptr(int32(1)),
+					ReportTime:     timestamppb.New(time.Now().UTC().Truncate(time.Millisecond)),
+					Metrics:        checkpoint.Training.ValidationMetrics,
+				},
+			})
+			require.NoError(t, err)
+
+			_, err = api.GetCheckpoint(ctx, &apiv1.GetCheckpointRequest{
```
Comment (erikwilson) [inline master/internal/api_checkpoint_intg_test.go:422]: small nit: would be good to verify the checkpoint too

Comment (AmanuelAaron) [reply master/internal/api_checkpoint_intg_test.go]: Added

File: master/internal/api_checkpoint.go [hunk 2]
```diff
@@ -129,12 +129,13 @@ func (a *apiServer) GetCheckpoint(
 	}
 
 	resp := &apiv1.GetCheckpointResponse{}
+	resp.Checkpoint = &checkpointv1.Checkpoint{}
 
-	ckpt, err := internaldb.GetCheckpoint(ctx, req.CheckpointUuid)
-	if err != nil {
-		return resp, errors.Wrapf(err, "error fetching checkpoint %s from database", req.CheckpointUuid)
+	if err := a.m.db.QueryProto(
```
Comment (salonig23) [inline master/internal/api_checkpoint.go:134]: Can we add a comment here saying not to use Bun?

**Review by salonig23 — APPROVED** [review]
LGTM!

File: master/internal/api_checkpoint.go (hunk 2 above)
Comment (AmanuelAaron) [reply master/internal/api_checkpoint.go:136]: Added. Thanks!

**Review by erikwilson — APPROVED** [review]
🎉

**Pull request merged**